### PR TITLE
CCIP-3862 Add tokenDestGasOverhead to CalculateMessageMaxGas

### PR DESCRIPTION
**Pull request by rstout**

### CODE REVIEW

File: core/capabilities/ccip/ccipevm/gas_helpers.go [hunk 1]
```diff
@@ -85,5 +94,6 @@ func (gp EstimateProvider) CalculateMessageMaxGas(msg cciptypes.Message) uint64
 		SupportsInterfaceCheck +
```
Comment (0xnogo) [inline core/capabilities/ccip/ccipevm/gas_helpers.go:104]: do we need to also add the `destGasOverhead`?

https://github.com/smartcontractkit/chainlink/blob/23452266132228234312947660374fb393e96f1a/contracts/src/v0.8/ccip/FeeQuoter.sol#L97

Comment (rstout) [reply core/capabilities/ccip/ccipevm/gas_helpers.go:104]: Is this a priority @matYang? I'll add a TODO and create a ticket in the meantime.

File: core/capabilities/ccip/ccipevm/gas_helpers.go [hunk 2]
```diff
@@ -51,18 +51,27 @@ func bytesForMsgTokens(numTokens int) int {
 }
 
 // CalculateMessageMaxGas computes the maximum gas overhead for a message.
+// TODO: update interface to return error
```
Comment (winder) [inline core/capabilities/ccip/ccipevm/gas_helpers.go]: Might be worth updating now to simplify migration in the future. You'd have to come up with a second function name, which isn't ideal.

i.e. leave a shim to delete later
```
func (gp EstimateProvider) CalculateMessageMaxGas(msg cciptypes.Message) uint64 {
    max, err := gp.CalculateMessageMaxGasWithError(msg)
    if err != nil {
        panic(err)
    }
    return max
}
```

**Review by winder — DISMISSED** [review]

File: core/capabilities/ccip/ccipevm/gas_helpers.go (hunk 2 above)
Comment (mateusz-sekara) [reply core/capabilities/ccip/ccipevm/gas_helpers.go]: Agree, if it's simple maybe it's gonna better to it right now. If not then leave a ticket in our backlog, it's easier to track it down if keep it in Jira as well

**Review by mateusz-sekara — DISMISSED** [review]

**Review by winder — APPROVED** [review]

File: core/capabilities/ccip/ccipevm/gas_helpers.go [hunk 3]
```diff
@@ -52,15 +53,33 @@ func bytesForMsgTokens(numTokens int) int {
...
+	if err != nil {
+		panic(err)
+	}
+	return maxGas
+}
+
+// CalculateMessageMaxGasWithError computes the maximum gas overhead for a message.
+// TODO: Add destGasOverhead, see:
+// https://github.com/smartcontractkit/chainlink/blob/23452266132228234312947660374fb393e96f1a/contracts/src/v0.8/ccip/FeeQuoter.sol#L97
+func (gp EstimateProvider) CalculateMessageMaxGasWithError(msg cciptypes.Message) (uint64, error) {
```
Comment (patrickhuie19) [inline core/capabilities/ccip/ccipevm/gas_helpers.go:66]: `CalculateMessageMaxGasWithError` naming implies to me that we are doing something useful with the error. Panic'ing is ...useful, just not the usual application. I see both API's are public and exportable - is there a plan to deprecate `CalculateMessageMaxGas`

**Review by patrickhuie19 — APPROVED** [review]

**Review by vyzaldysanchez — APPROVED** [review]

**Pull request merged**